server/game_engine/GameEngine.js

```diff
@@ -176,16 +269,38 @@ class Game {
             blueGuessers: this.blueGuessers,
             blueSpymaster: this.blueSpymaster,
             redGuessers: this.redGuessers,
-            redSpymaster: this.redSpymaster
-        }
+            redSpymaster: this.redSpymaster,
+        };
     }
 
     assignRole(playerId, role) {
-        role.indexOf("Spymaster") === -1 ? this[role].push(playerId) : this[role] = playerId;
+        role.indexOf("Spymaster") === -1
+            ? this[role].push(playerId)
+            : (this[role] = playerId);
+        this.playersReady.push(playerId);
+    }
+
+    removeRole(playerId, role) {
+        role.indexOf("Spymaster") === -1
+            ? (this[role] = this[role].filter((player) => player !== playerId))
+            : (this[role] = "");
```

Comment:
It looks like for a role like `Spymaster` the default value would be an empty string but for `blueGuessers` or `redGuessers`, it should be empty array []